### PR TITLE
Move methods from security managers to `FabAirflowSecurityManagerOverride`

### PR DESCRIPTION
**Pull request by vincbeck**
The intent of this PR is to move FAB related methods from security managers ([AirflowSecurityManager](https://github.com/apache/airflow/blob/main/airflow/www/security.py), [BaseSecurityManager](https://github.com/apache/airflow/blob/main/airflow/www/fab_security/manager.py) and [SecurityManager](https://github.com/apache/airflow/blob/main/airflow/www/fab_security/sqla/manager.py)) to `FabAirflowSecurityManagerOverride`. Since the effort is quite big, this PR is just a first step and more methods will be moved in separate PRs. This PR moves the most "obvious" methods to be moved. This will help to review the PR as well.

Also, in order to avoid having one gigantic `FabAirflowSecurityManagerOverride`, I decided to split it in different modules which `FabAirflowSecurityManagerOverride` inherits from. So far I created two modules: `DB` and `oauth`, each one defining methods related to either the database or Oauth.

There is no new code in this PR, the changes are only code being moved around.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -491,6 +491,7 @@ repos:
         exclude: >
           (?x)
           ^airflow/api_connexion/openapi/v1.yaml$|
+          ^airflow/auth/managers/fab/security_manager/|
```
Comment (vandonr-amz) [inline .pre-commit-config.yaml:502]: why do we need this when the previous location was not in this list ?

Comment (vincbeck) [reply .pre-commit-config.yaml:502]: It was: `^airflow/www/fab_security/manager.py$|`

Comment (vandonr-amz) [reply .pre-commit-config.yaml:502]: ah yes, I thought it was in alphabetical order and I missed it -_-

**Review by vandonr-amz — APPROVED** [review]
looks OK from what I can tell

**Review by o-nikolas — APPROVED** [review]
Can be tricky to review these types of large PRs, but from what I can tell it LGTM :+1:

**Comment by potiuk** [comment]
I am sorry - I have just merged https://github.com/apache/airflow/pull/33347 which has been important to be out for 2.7.0, and this one conflicts with it so if you please rebase witht the chnage of mine and ping me, I will review it quickly (also now when I am more familiar with recent changes you've done). Sorry - but 2.7. has been priority over the last few days

**Comment by vincbeck** [comment]
Totally understandable 👌  No worries @potiuk. I just resolved the conflicts. If you can review it, that would be awesome 🥳

**Review by uranusjr — APPROVED** [review]
I did not compare the code in detail and trust you just moved them.

**Pull request merged**